gammapy/background/energy_offset_array.py

```diff
@@ -178,9 +197,93 @@ def bin_volume(self):
         TODO: explain with formula and units
         """
         delta_energy = self.energy.bands
-        solid_angle = np.pi * (self.offset[1:] ** 2 - self.offset[:-1] ** 2)
+        solid_angle = self.solide_angle
         # define grid of deltas (i.e. bin widths for each 3D bin)
         delta_energy, solid_angle = np.meshgrid(delta_energy, solid_angle, indexing='ij')
         bin_volume = delta_energy * (solid_angle).to('sr')
 
         return bin_volume
+
+
+    def curve_at_energy(self, energy):
+        """
+        Parameters
+        ----------
+        energy: `~astropy.units.Quantity`
+
+        Returns
+        -------
+
+        """
+        table=Table()
+        table["offset"]=self.offset_bin_center
+        table["1Dcurve"]=self.evaluate(energy, offset=None)
+        return table
+
+    def curve_at_offset(self, offset):
+        """
+
+        Parameters
+        ----------
+        offset: `~astropy.coordinates.Angle`
+
+        Returns
+        -------
+
+        """
+        table=Table()
+        table["energy"]=self.energy.log_centers
+        table["1Dcurve"]=self.evaluate(None, offset)
+        return table
+
+
+    def acceptance_curve_in_energy_band(self, energy_band, energy_bins=10):
+        """
+        Parameters
+        ----------
+        energy_band : `~astropy.units.Quantity`
+            Tuple ``(energy_min, energy_max)``
+        energy_bins : int or `~astropy.units.Quantity`
+            Energy bin definition.
+
+        """
+        [Emin,Emax]=energy_band
+        energy_edges=EnergyBounds.equal_log_spacing(Emin, Emax, energy_bins)
+        energy_bins=energy_edges.log_centers
+        acceptance =+ self.evaluate(energy_bins, offset=None)
+        #this is super crado to do that, is there not a way to directly multiply 2D(N1, N2) with 1D(N1)?
+        acceptance_tot=np.zeros(len(self.offset) - 1)
+        for i in range(len(energy_bins)):
+            acceptance_tot += acceptance[i,:]*self.solide_angle.to('sr')*energy_edges.bands[i].to('MeV')
+        table=Table()
+        table["offset"]=self.offset_bin_center
+        table["Acceptance"]=acceptance_tot
+        return table
+
+
+
+    def to_multi_Cube(self, Cube):
+        """
+
+        Parameters
+        ----------
+        Cube: `~gammapy.background.models.CubeBackgroundModel`
+
+        Returns
+        -------
+
+        """
+        coordx_center=(Cube.counts_cube.coordx_edges[:-1]+Cube.counts_cube.coordx_edges[1:])/2.
+        coordy_center=(Cube.counts_cube.coordy_edges[:-1]+Cube.counts_cube.coordy_edges[1:])/2.
+
+        xx, yy = np.meshgrid(coordx_center, coordy_center)
+        dist=np.sqrt(xx**2 + yy**2)
+        shape=np.shape(dist)
+        data=self.evaluate(self.energy.log_centers, dist.flat)
```

Comment:
pb ici of out of bounds for the interpolator since dist has values larger than self.offset. I don't understand why he can't extrapolated for this values!